### ISSUE TITLE
Is the issue of non-reproducible results in keras not resolved yet?

### ISSUE THREAD

**Issue by ddolgov** [fchollet/keras#6737]
This is related to issues https://github.com/fchollet/keras/issues/439, https://github.com/fchollet/keras/issues/2280 and https://github.com/fchollet/keras/issues/2743

I am unable to reproduce my results from run to run with either tensorflow or theano backend (running on CPU) and I do call `np.random.seed(11)` before importing keras.
```
>>> theano.__version__
'0.9.0.dev-d18ce33b02273026ea7986fe9f9878546db6e8cf'
>>> tensorflow.__version__
'1.1.0'
>>> np.__version__
'1.12.1'
>>> keras.__version__
'2.0.4'
```

**Comment by StefanoD** [comment]
As far as I can, this is an issue of the backends. If you would use them directly, you would have the same issue.

**Comment by thaolmk54** [comment]
How to claim the result if TF backend is non-deterministic as it produces different result after each run?

**Comment by VanitarNordic** [comment]
it has no resolved yet, very bad

**Comment by ng0323** [comment]
Refering to the CIFAR10 CNN training example in Keras.
How can I get determinism with Theano backend? I already set the random seed as above.
I used Theano backend without cuDNN (because my cuDNN version is too old), but still unable to get determinism. I know that cuDNN has non-determinism.
What else could be the source of non-determinism?
